docs/interpreter/spark.md

```diff
@@ -343,6 +343,29 @@ myScalaDataFrame = DataFrame(z.get("myScalaDataFrame"), sqlContext)
   </div>
 </div>
 
+#### Alternative Object Retrieval Mechanism
+In addition to the programmatic access using `z.get()` shown above, 
+some interpreters also support inline interpolation of such objects
+into command lines by using substitution patterns of the form `{object-key}`.
+The following example shows how the value of an object `put` into `ZeppelinContext` 
+by the Scala interpreter is spliced into a command of the SQL interpreter.
+
+__In Scala interpreter:__
+```
+val tableName = ...
+z.put("tableName", tableName)
+```
+__In SQL interpreter:__
+```
+select * from {tableName}
+```
+The escaping pattern `{{anything}}` may be used in situations where `{` and `}` are needed 
+in the command line. The pattern `{{anything}}` is translated into `{anything}` and spliced 
+into the containing command.
+
+The use of the `{object-key}` format for object interpolation is currently only available 
+in the SQL interpreter, but will be provided in other interpreters progressively.
```

Comment:
I'm a bit confused which commits are going to which. There seems to be more explanation in other commits that is not showing up in this PR.

but as discussed, let's explain more the behavior. It's fine to say this is modeled on Jupyter but different in these ways. eg. only when object-key is found in context, etc. I see some of these in the other commit.

Reply:
I think we need quite a bit in the documentation to explain this more.